code/game/machinery/mind_machine.dm

```diff
@@ -0,0 +1,846 @@
+				woopsTarget = pick(woopsOne)
+				var/list/M_spells = list()
+				M_spells = M.spell_list.Copy()
+				if(M.mind.special_verbs.len)	//All the safeties in higher swap. Prevents people from getting cult spells via swap
+					for(var/V in M.mind.special_verbs)
+						M.verbs -= V
+				for(var/spell/S in M.spell_list)
+					M.remove_spell(S)
+				M.mind.transfer_to(woopsTarget)
+				for(var/spell/S in M_spells)
+					woopsTarget.add_spell(S)
+				to_chat(M, "<span class='danger'>Your mind is thrown out of the machine and forced into a nearby vessel</span>")
+				playsound(M, "sound/effects/phasein.ogg", 50, 1)
+				errorMessage = "Error in mind delivery"
+			else
+				to_chat(M, "<span class='danger'>Your mind is severely damaged by the feedback</span>")
+				playsound(M, "sound/misc/balloon_pop.ogg", 50, 1)
+				if(ishuman(M))
+					M.adjustBrainLoss(75)
+				errorMessage = "Mind sustained damage"
+		if(mindTypeTwo == "Higher" || mindTypeTwo == "Artificial")
+			var/list/woopsTwo = list()
+			M = occupantTwo
+			for(var/mob/living/R in mob_list)
+				if((!R.mind) && (R.stat != 2) && (get_dist(src, R) <50) && (connectTwo.z == R.z) && (!is_type_in_list(R, illegalSwap)))
+					woopsTwo += R
+			if(woopsTwo.len)
+				var/list/M_spells = list()
+				M_spells = M.spell_list.Copy()
+				woopsTarget = pick(woopsTwo)
+				if(M.mind.special_verbs.len)
+					for(var/V in M.mind.special_verbs)
+						M.verbs -= V
+				for(var/spell/S in M.spell_list)
+					M.remove_spell(S)
+				M.mind.transfer_to(woopsTarget)
+				for(var/spell/S in M_spells)
+					woopsTarget.add_spell(S)
+				to_chat(M, "<span class='danger'>Your mind is thrown out of the machine and forced into a nearby vessel</span>")
+				playsound(M, "sound/effects/phasein.ogg", 50, 1)
+				errorMessage = "Error in mind delivery"
+			else
+				to_chat(M, "<span class='danger'>Your mind is severely damaged by the feedback</span>")
+				playsound(M, "sound/misc/balloon_pop.ogg", 50, 1)
+				if(ishuman(M))
+					M.adjustBrainLoss(75)
+				errorMessage = "Mind sustained damage"
+		if(mindTypeOne == "Lower")
+			M = occupantOne
+			M.faction = pick(randFaction)
+		if(mindTypeTwo == "Lower")
+			M = occupantTwo
+			M.faction = pick(randFaction)
+	else
+		soulShardSafety = FALSE
+		errorMessage = "Safety consumed"
+	currentlySwapping = FALSE
+	occupantScan = FALSE
+	connectOne.icon_state = "mind_pod_closed"
+	connectTwo.icon_state = "mind_pod_closed"
+	icon_state = "mind_hub"
+	swapProgress = 0
+
+/obj/machinery/mind_machine/mind_machine_hub/proc/theFlySwap(var/mob/living/MO, var/mob/living/MT)
+	var/list/mindFly = list()
+	var/list/simpFly = list()
+	MO = occupantOne
+	MT = occupantTwo
+	for(var/obj/item/weapon/holder/D in theFly)
+		for(var/mob/M in D.contents)
+			theFly += M //Gets the mob stored in the holder, adds it to the list
+		D.Destroy() //Turns them from holder to mob
+	theFly += MO
+	theFly += MT
+	for(var/mob/living/q in theFly)
+		if(q.mind)
+			mindFly += q
+			theFly -= q
```

Comment:
`theFly` isn't used for the rest of this method until it's reset so there's no use to explicitly removing items from it like this.